src/List.js

```diff
@@ -371,6 +379,9 @@ export function emptyList() {
 
 function updateList(list, index, value) {
   index = wrapIndex(list, index);
+  if (index !== index) {
+    throw new Error('Cannot set item at unknown index "' + index + '" on ' + listName(list));
```

Comment:
No other set method throws like this at the moment, so this would be a breaking change. I think this may be the right thing to do long term, but for now, let's make it silent.